compair/models/comparison.py

```diff
@@ -118,6 +119,27 @@ def _get_new_comparison_pair(cls, course_id, assignment_id, user_id, pairing_alg
             )) \
             .all()
 
+        answers_with_criterion_score = Answer.query \
+            .with_entities(Answer, AnswerCriterionScore.criterion_id, AnswerCriterionScore.score) \
+            .join(AnswerCriterionScore) \
+            .filter(and_(
+                Answer.user_id.notin_(ineligible_user_ids),
+                Answer.assignment_id == assignment_id,
+                Answer.active == True,
+                Answer.practice == False,
+                Answer.draft == False
+            )) \
+            .all()
+
+        assignment_criterion_weights = Assignment.query \
```

Comment:
you can probably just query `AssignmentCriterion` with assignment_id = assignment_id instead of doing a join